src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -1,17 +1,121 @@
+    public String greeting(@RequestParam(name = "name", required = false, defaultValue = "Poli") String name,
+            Model model) {
         model.addAttribute("name", name);
         return "greeting";
     }
 
+    /**
+     * @return all the books contained in the model
+     */
+    @GetMapping("/all")
+    public Optional<List<Book>> findAll() {
+        List<Book> allBooks = new ArrayList<>();
+        if (bookRepository.findAll().iterator().hasNext()) {
+            bookRepository.findAll().forEach(b -> allBooks.add(b));
+            return Optional.of(allBooks);
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    /**
+     * @param bookAuthor indicates the author to be searched
+     *
+     * @return the entire book of the selected author
+     */
+    @GetMapping("/author/{bookAuthor}")
+    public Optional<List<Book>> findByAuthor(@PathVariable String bookAuthor) {
+        List<Book> authorBooks = new ArrayList<>();
+        if (bookRepository.findByBookAuthor(bookAuthor).iterator().hasNext()) {
+            bookRepository.findByBookAuthor(bookAuthor).forEach(b -> authorBooks.add(b));
+            return Optional.of(authorBooks);
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    @GetMapping("/{bookGenre}")
+    public Optional<List<Book>> findByGenre(@PathVariable String bookGenre) {
+        List<Book> genreBooks = new ArrayList<>();
+        if (bookRepository.findByBookGenre(bookGenre).iterator().hasNext()) {
+            bookRepository.findByBookGenre(bookGenre).forEach(b -> genreBooks.add(b));
+            return Optional.of(genreBooks);
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    /**
+     * @param book indicates the book to be added
+     *
+     * @return the created book
+     */
+    @PostMapping("/create")
+    @ResponseStatus(HttpStatus.CREATED)
+    public Book create(@RequestBody Book book) {
+        return bookRepository.save(book);
+    }
+
+    /**
+     * @param id indicates the id of the book to be searched
+     *
+     * @throws BookNotFoundException returns an exception if the book is not found
+     */
+    @DeleteMapping("/{id}")
+    @ResponseStatus(HttpStatus.OK)
+    public void delete(@PathVariable Long id) {
+        bookRepository.findById(id)
+                .orElseThrow(BookNotFoundException::new);
+        bookRepository.deleteById(id);
+    }
+
+    /**
+     * @param id   indicates the id of the book to be updated
+     * @param book indicates the information of the book to be updated
+     *
+     * @return the updated book
+     * @throws BookNotFoundException returns an exception if the book is not found
+     */
+    @PutMapping("/{id}")
+    @ResponseStatus(HttpStatus.OK)
+    public Book updateBook(@RequestBody Book book, @PathVariable Long id) {
+        bookRepository.findById(id)
+                .orElseThrow(BookNotFoundException::new);
+        return bookRepository.save(book);
```

Comment:
Nice job. Now, what would happen if the id from the path is not the same as the id from the request body? you have to avoid this behavior.